freshservice/custom_objects.go

```diff
@@ -18,34 +16,64 @@ type CustomObjectService struct {
 	client *Client
 }
 
-// CustomObjects contains Collection an array of CustomObject
-type CustomObjects struct {
-	Collection []CustomObject `json:"custom_objects"`
+type CustomObject struct {
+	ID          int      `json:"id,omitempty"`
+	Name        string   `json:"name,omitempty"`
+	Title       string   `json:"title,omitempty"`
+	Description string   `json:"description,omitempty"`
+	Fields      []Fields `json:"fields,omitempty"`
+	Meta        int      `json:"total_records_count,omitempty"`
 }
 
-// CustomObjectWrapper contains Details of a CustomObject
-type CustomObjectWrapper struct {
-	Details CustomObject `json:"custom_object"`
+type Fields struct {
+	Name     string   `json:"name,omitempty"`
+	Label    string   `json:"label,omitempty"`
+	Type     string   `json:"type,omitempty"`
+	Required bool     `json:"required,omitempty"`
+	Choices  []string `json:"choices,omitempty"`
+	Meta     string   `json:"meta,omitempty"`
 }
 
-type CustomObject struct {
-	ID          int       `json:"id,omitempty"`
-	Title       string    `json:"title,omitempty"`
-	Description string    `json:"description,omitempty"`
-	UpdatedAt   time.Time `json:"updated_at,omitempty"`
-	//LastUpdatedBy
-	//Meta
+// CustomObjectRecord contains a single custom object record
+type CustomObjectRecord struct {
+	CreatedAt      time.Time `json:"bo_created_at,omitempty"`
+	CreatedBy      CreatedBy `json:"bo_created_by,omitempty"`
+	DisplayId      int       `json:"bo_display_id,omitempty"`
+	UpdatedAt      time.Time `json:"bo_updated_at,omitempty"`
+	UpdatedBy      UpdatedBy `json:"bo_updated_by,omitempty"`
+	DataSource     string    `json:"data_source_name,omitempty"`
+	DataSourceType string    `json:"data_source_id,omitempty"`
+}
+
+type CreatedBy struct {
+	Id    int    `json:"id,omitempty"`
+	Value string `json:"value,omitempty"`
 }
 
-type LastUpdatedBy struct {
-	Email  string `json:"email,omitempty"`
-	ID     int    `json:"id,omitempty"`
-	System string `json:"name,omitempty"`
+type UpdatedBy struct {
+	ID    int    `json:"id,omitempty"`
+	Value string `json:"value,omitempty"`
 }
-type Meta struct {
-	Count   int `json:"count,omitempty"`
-	Page    int `json:"page,omitempty"`
-	PerPage int `json:"per_page,omitempty"`
+
+// CustomObjectRecords contains an array of CustomObject
+type CustomObjectRecords struct {
+	Collection   []CustomObjectRecordWrapper `json:"records,omitempty"`
+	NextPageLink string                      `json:"next_page_link,omitempty"`
+	Meta         map[string]interface{}      `json:"meta,omitempty"`
+}
+
+// CustomObjects contains Collection an array of CustomObject
+type CustomObjects struct {
+	Collection []CustomObject `json:"custom_objects"`
+}
+
+//CustomObjectRecordWrapper contains Details of a CustomObjectRecords
+type CustomObjectRecordWrapper struct {
+	Data CustomObjectRecord `json:"data,omitempty"`
+}
+
+type CustomObjectRecordUpdate struct {
+	CustomObjectRecord CustomObjectRecord `json:"custom_object,omitempty"`
```

Comment:
I'm surprised we don't need to wrap this field, does he inner object have a `data` field?

Reply:
we have a CustomObjectRecordWrapper